src/iptvsimple/utilities/StreamUtils.cpp

```diff
@@ -63,7 +63,56 @@ void StreamUtils::SetAllStreamProperties(PVR_NAMED_VALUE* properties, unsigned i
     }
     else // inputstream.adpative
     {
-      StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
+      // If no media headers are explicitly set for inputstream.adaptive,
+      // strip the headers from streamURL and put it to media headers property
+
+      bool haveStreamHeaders = !channel.GetProperty("inputstream.adaptive.stream_headers").empty();
+      if (!haveStreamHeaders)
+      {
+        // preserve any kodi protocol options after "|"
+        std::string url = streamURL;
+        std::string protocolOptions;
+        std::string untrimmed_header = "";
+
+        size_t found = streamURL.find_first_of('|');
+        if (found != std::string::npos)
+        {
+          url = streamURL.substr(0, found);
+          protocolOptions = streamURL.substr(found + 1, streamURL.length());
+
+          std::vector<std::string> headers = StringUtils::Split(protocolOptions, "&");
+          protocolOptions.clear();
+          for (std::vector<std::string>::iterator b(headers.begin()), e(headers.end()); b != e; ++b)
```

Comment:
why not just `for (std::string header : headers)` and then just find the `=` in each header?

Much easier to read and understand.

Reply:
I also thought of AddHeader modification in the meantime. Let me run it quickly and see if code gets simpler :)